src/main/java/org/pulseus/auth/service/rest/controller/CSVController.java

```diff
@@ -0,0 +1,114 @@
+import com.liferay.portal.kernel.util.StringPool;
+
+@RestController
+public class CSVController {
+	Configuration configuration = ConfigurationFactoryUtil.getConfiguration(PortalClassLoaderUtil.getClassLoader(), "portlet");
+
+	private static final Logger log = LogManager.getLogger(JWTController.class);
+
+	@RequestMapping(value = "/addusers", method = RequestMethod.GET)
+	public void addusers() throws Exception {
+		System.out.println("adding users");
+		ServiceContext serviceContext = new ServiceContext();
+		Long companyId = CompanyThreadLocal.getCompanyId();	
+
+
+
+		BufferedReader br = null;
+		String line = "";
+		String cvsSplitBy = ",";
+
+		br = new BufferedReader(new FileReader("/opt/pulse/users.csv"));
+		while (true) {
+			try {
+				if ((line = br.readLine()) != null) {
+					// use comma as separator
+					String[] value = line.split(cvsSplitBy);
+					Role role = RoleLocalServiceUtil.getRole(companyId, value[5]);
+					Long roleId = 	role.getRoleId();
+
+					Organization org1 = OrganizationLocalServiceUtil.getOrganization(companyId, value[6]);
+					Long orgId1 = org1.getOrganizationId();
+
+					Organization org2 = OrganizationLocalServiceUtil.getOrganization(companyId, value[7]);
+					Long orgId2 = org2.getOrganizationId();
+
+					long creatorUserId = 0;
+					boolean autoPassword = false;
+					boolean autoScreenName = false;
+					long facebookId = 0;
+					String openId = StringPool.BLANK;
+					Locale locale = serviceContext.getLocale();
+					String middleName = StringPool.BLANK;
+					int prefixId = 0;
+					int suffixId = 0;
+					boolean male = true;
+					int birthdayMonth = Calendar.JANUARY;
+					int birthdayDay = 1;
+					int birthdayYear = 1970;
+					String jobTitle = StringPool.BLANK;
+					long[] groupIds = null;
+					long[] organizationIds = {orgId1,orgId2};
+					long[] roleIds = {roleId};
+					long[] userGroupIds = null;
+					boolean sendEmail = false;
+
+					String screenName = value[0];
+					String emailAddress = value[1];
+					String firstName = value[2];
+					String lastName = value[3];
+					String password1 = value[4];
+					String password2 = value[4];
+
+					String uuid = UUID.randomUUID().toString().replace("-", "");
+					serviceContext.setUuid(uuid);
+
+					User user = UserLocalServiceUtil.addUser(
+							creatorUserId, companyId, autoPassword, password1, password2,
+							autoScreenName, screenName, emailAddress, facebookId, openId,
+							locale, firstName, middleName, lastName, prefixId, suffixId, male,
+							birthdayMonth, birthdayDay, birthdayYear, jobTitle, groupIds,
+							organizationIds, roleIds, userGroupIds, sendEmail, serviceContext);
+
+					user = UserLocalServiceUtil.updateEmailAddressVerified(user.getUserId(), true);
+					user = UserLocalServiceUtil.updatePasswordReset(user.getUserId(), false);
+
+					System.out.println("user added");
+					
+				}
+			}
+
+			catch(Exception e) {
+				System.out.println(e.getMessage());
```

Comment:
I'm assuming this is the part where things like duplicate usernames would be caught and outputted, right? Have you tested it to see what it does? I'm guessing that, at a minimum, the "line that has the error" should be added to this message so an admin could do something with it. They could be collected as well and returned as part of the GET response.

Reply:
Yes, I tested it. It logs that a duplicate user already exists by the given name in CSV. Would adding e.printStackTrace() suffice?